metronome/hotstuff/service/src/io/iohk/metronome/hotstuff/service/messages/DuplexMessage.scala

```diff
@@ -0,0 +1,24 @@
+package io.iohk.metronome.hotstuff.service.messages
+
+import io.iohk.metronome.hotstuff
+import io.iohk.metronome.hotstuff.consensus.basic.Agreement
+
+/** Messages type to use in the networking layer if the use case has
+  * application specific message types, e.g. ledger synchronisation,
+  * not just the general BFT agreement (which could be enough if
+  * we need to execute all blocks to synchronise state).
+  */
+sealed trait DuplexMessage[A <: Agreement, M]
+
+object DuplexMessage {
+
+  /** General BFT agreement message. */
+  case class AgreementMessage[A <: Agreement](
+      message: hotstuff.service.messages.HotStuffMessage[A]
```

Comment:
Do we need a qualified name here?

Reply:
Not really, just left it for consistency with what the `HotStuffMessage` itself looks like, where it is needed for `SyncMessage` but increases the readability for the `Message` as well. Can remove it if you prefer.